### PR TITLE
[FEATURE] DataDocs Renderer for `ExpectMulticolumnSumToEqual` and `LikePattern` Expectations

### LINKED ISSUE #7453: [BUG] expect_multicolumn_sum_to_equal outputs AttributeError / TypeError
**Describe the bug**
When using expect_multicolumn_sum_to_equal, it does not outputs the result as intended. Instead, it outputs an error which fails to show up in Data doc anyway.

**To Reproduce**
using expect_multicolumn_sum_to_equal 

**Expected behavior**
This should work as intended as per document, where multi columns summation should result in a value

**Environment (please complete the following information):**
 - Operating System: [e.g. Linux, MacOS, Windows]: MacOS
 - Great Expectations Version: [e.g. 0.13.2]: 0.16.1

**Additional context**

image

### PR DESCRIPTION
**Pull request by Shinnnyshinshin**
Closes #7453 
Follow up to #8745 

## What has changed? 
Adding `prescriptive_renderer` to the following Expectations:
  * `expect_multicolumn_sum_to_equal`
  * `expect_column_values_to_match_like_pattern`
  * `expect_column_values_to_not_match_like_pattern`
  * `expect_column_values_to_match_like_pattern_list`
  * `expect_column_values_to_not_match_like_pattern_list`

Enabled test that checked whether all Expectations would output proper BulletListContentBlock, which is downstream of the `prescriptive_renderer`

## Example of rendered output
Screenshot 2023-09-22 at 10 18 56 AM
Screenshot 2023-09-21 at 5 58 55 PM

## Example of rendered output with `mostly`
Screenshot 2023-09-22 at 10 17 57 AM

Screenshot 2023-09-21 at 5 57 07 PM

- [x] Description of PR changes above includes a link to [an existing GitHub issue](https://github.com/great-expectations/great_expectations/issues)
- [x] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]
- [x] Code is linted - run `invoke lint` (uses `black` + `ruff`)
- [x] Appropriate tests and docs have been updated

For more information about contributing, see [Contribute](https://docs.greatexpectations.io/docs/contributing/contributing_checklist).

After you submit your PR, keep the page open and **monitor the statuses of the various checks made by our continuous integration process at the bottom of the page. Please fix any issues that come up** and [reach out on Slack](https://greatexpectations.io/slack) if you need help. Thanks for contributing!

### CODE REVIEW

File: great_expectations/expectations/core/expect_column_values_to_match_like_pattern.py
Comment (cdkini) [inline great_expectations/expectations/core/expect_column_values_to_match_like_pattern.py]: Could we bring back the enum?

File: great_expectations/expectations/core/expect_column_values_to_match_like_pattern.py
Comment (cdkini) [inline great_expectations/expectations/core/expect_column_values_to_match_like_pattern.py:135]: Does this deviate at all from the parent class signature?

File: great_expectations/expectations/core/expect_column_values_to_match_like_pattern.py
Comment (cdkini) [inline great_expectations/expectations/core/expect_column_values_to_match_like_pattern.py]: Probably a larger issue with how we write these renderers in general but I'd love to move away from magic strings and numbers by providing context through constants and enums. Not a blocker just a thought.

File: great_expectations/expectations/core/expect_column_values_to_match_like_pattern.py
Comment (cdkini) [inline great_expectations/expectations/core/expect_column_values_to_match_like_pattern.py:168]: So a list with a single element? Is there a reason we don't just return the element? Do other renderers have lists with more values?

File: great_expectations/expectations/core/expect_column_values_to_match_like_pattern_list.py
Comment (cdkini) [inline great_expectations/expectations/core/expect_column_values_to_match_like_pattern_list.py]: Is precision always 15? If so, maybe we can make that a constant to share across the different impls?

File: great_expectations/expectations/core/expect_column_values_to_not_match_like_pattern.py
Comment (cdkini) [inline great_expectations/expectations/core/expect_column_values_to_not_match_like_pattern.py:134]: Do we have tests for these new renderer implementations? I remember having some kind of snapshot test harness to ensure the outputs of these were covered?

File: great_expectations/expectations/core/expect_multicolumn_sum_to_equal.py
Comment (cdkini) [inline great_expectations/expectations/core/expect_multicolumn_sum_to_equal.py:105]: Same note about enum here and anywhere else applicable

File: tests/render/test_render_BulletListContentBlock.py
Comment (cdkini) [inline tests/render/test_render_BulletListContentBlock.py:96]: Love to see this go 🙇🏽

**Review by cdkini — COMMENTED** [review]
I'll let someone from Zelda approve but LGTM - just some minor nits

**Review by anthonyburdi — APPROVED** [review]
LGTM pending resolution of Chetan's comments.

File: great_expectations/expectations/core/expect_column_values_to_not_match_like_pattern.py
Comment (Shinnnyshinshin) [reply great_expectations/expectations/core/expect_column_values_to_not_match_like_pattern.py:134]: This is currently tested by the test in : `tests/render/test_render_BulletListContentBlock.py` which goes through each of the JSON test definitions and ensures that a proper rendered output is generated.

File: great_expectations/expectations/core/expect_column_values_to_match_like_pattern.py
Comment (Shinnnyshinshin) [reply great_expectations/expectations/core/expect_column_values_to_match_like_pattern.py]: done

File: great_expectations/expectations/core/expect_column_values_to_match_like_pattern.py
Comment (Shinnnyshinshin) [reply great_expectations/expectations/core/expect_column_values_to_match_like_pattern.py:135]: This corrects the deviation. The parent class's signature is the `List[RenderedStringTemplateContent]`

https://github.com/great-expectations/great_expectations/blob/72fe5b1453aee72a27e790035845a94ba9428980/great_expectations/expectations/expectation.py#L541

File: great_expectations/expectations/core/expect_column_values_to_match_like_pattern.py
Comment (Shinnnyshinshin) [reply great_expectations/expectations/core/expect_column_values_to_match_like_pattern.py:168]: good point. This may be worth refactoring

File: great_expectations/expectations/core/expect_column_values_to_match_like_pattern_list.py
Comment (Shinnnyshinshin) [reply great_expectations/expectations/core/expect_column_values_to_match_like_pattern_list.py]: This was sortof already done, but I think it can be cleaned up as part of the PR. Thank you for the suggestion here
https://github.com/great-expectations/great_expectations/blob/d9b8180fece6e285f70e0df20c295612b23a31d1/great_expectations/render/util.py#L18

**Pull request merged**
